presto-main/src/main/java/com/facebook/presto/sql/planner/PlanFragmenter.java

```diff
@@ -248,7 +248,7 @@ private SubPlan analyzeGroupedExecution(Session session, SubPlan subPlan, boolea
     {
         PlanFragment fragment = subPlan.getFragment();
         GroupedExecutionProperties properties = fragment.getRoot().accept(new GroupedExecutionTagger(session, metadata, nodePartitioningManager), null);
-        if (properties.isSubTreeUseful()) {
+        if (properties.isSubTreeUseful() && !(fragment.getPartitioning().getConnectorHandle() instanceof SystemPartitioningHandle)) {
```

Comment:
Personally I would prefer to disable the grouped execution for local query runner explicitly (or at least explicitly check for the `forceSinceNode` flag).

Reply:
updated to check the forceSingleNode flag